module/zfs/zfs_ioctl.c

```diff
@@ -4605,6 +4623,29 @@ zfs_ioc_userspace_upgrade(zfs_cmd_t *zc)
 	return (error);
 }
 
+/*
+ * inputs:
+ * zc_name		name of filesystem
+ *
+ * outputs:
+ * none
+ */
+static int
+zfs_ioc_userobjspace_upgrade(zfs_cmd_t *zc)
+{
+	int err;
+	zfs_sb_t *zsb;
+
+	err = zfs_sb_hold(zc->zc_name, FTAG, &zsb, B_FALSE);
+	if (err != 0)
```

Comment:
I don't believe there's any need to mount the filesystem in order to upgrade it.  If it is mounted that's fine, otherwise taking on a hold on the objset should be sufficient.  This function should be almost identical to the existing `zfs_ioc_userspace_upgrade()` function which does the same thing.